pilot/pkg/status/collections.go

```diff
@@ -66,21 +68,27 @@ func (s *StatusCollections) SetQueue(queue status.Queue) []krt.Syncer {
 }
 
 // registerStatus takes a status collection and registers it to be managed by the status queue.
-func registerStatus[I controllers.Object, IS any](c *Controller, statusCol krt.StatusCollection[I, IS]) {
-	reg := func(statusWriter status.Queue) krt.HandlerRegistration {
+func RegisterStatus[I controllers.Object, IS any](s *StatusCollections, statusCol krt.StatusCollection[I, IS]) {
+	reg := func(statusWriter Queue) krt.HandlerRegistration {
 		h := statusCol.Register(func(o krt.Event[krt.ObjectWithStatus[I, IS]]) {
 			l := o.Latest()
-			enqueueStatus(statusWriter, l.Obj, &l.Status)
+			status := &l.Status
+			if o.Event == controllers.EventDelete {
+				// if the object is being deleted, we should not reset status
+				var empty IS
+				status = &empty
+			}
+			enqueueStatus(statusWriter, l.Obj, status)
```

Comment:
@howardjohn made this change so that we correctly reset the status of objects that have been unselected from the collection, let me know what you think